tests/returntypes/autograd/test_autograd_qnode_new.py

```diff
@@ -871,8 +905,8 @@ def cost(a, b, c, weights):
 
     def test_second_derivative(self, interface, dev_name, diff_method, mode, tol):
         """Test second derivative calculation of a scalar valued QNode"""
-        if diff_method in {"adjoint"}:
-            pytest.skip("Test does not support adjoint.")
+        if diff_method not in {"parameter-shift", "backprop"}:
```

Comment:
I agree that the conditionals here get a bit messy. Overall I think `diff_method`s are not added very often, but it will become progressively harder to do so.
I think the supported configurations doc, together with its tests in `test_supported_confs` is great, because if a method does not appear there, there is a hint that coverage maybe was not investigated and the method might not have feature parity.

We should think about how to better structure these tests, and whether we can get closer to more modular unit tests.